diff --git a/aosp/dynamic_partition_control_android.cc b/aosp/dynamic_partition_control_android.cc
--- a/aosp/dynamic_partition_control_android.cc
+++ b/aosp/dynamic_partition_control_android.cc
@@ -917,6 +917,14 @@ bool DynamicPartitionControlAndroid::CheckSuperPartitionAllocatableSpace(
   }
 
   uint64_t full_space = builder->AllocatableSpace();
+  // Darth9 - allow allocatable space to be overridden
+  uint64_t new_size = full_space;
+  std::string new_size_prop = android::base::GetProperty("ro.dynamic.full_size", "");
+  if (!new_size_prop.empty() && (std::istringstream(new_size_prop) >> new_size) && new_size > full_space) {
+	LOG(INFO) << "DEBUG - replacing: " << full_space << " with: " << new_size;
+	full_space = new_size;
+  }
+  // :Darth9
   uint64_t half_space = full_space / 2;
   constexpr const char* fmt =
       "The maximum size of all groups for the target slot (%" PRIu64
